src/software/ai/evaluation/keep_away.cpp

```diff
@@ -0,0 +1,59 @@
+#include "software/ai/evaluation/keep_away.h"
+
+#include "software/ai/passing/cost_function.h"
+#include "software/math/math_functions.h"
+#include "software/optimization/gradient_descent_optimizer.h"
+
+double ratePasserPointForKeepAway(const Pass& pass, const Team& enemy_team)
+{
+    // the default values for these passing parameters
+    // TODO: cleanup passing parameters as part of #1987
+    // TODO: tweak all of these constants to get better behaviour
+    static constexpr double PASSER_ENEMY_PROXIMITY_IMPORTANCE   = 1.5;
+    static constexpr double RECEIVER_ENEMY_PROXIMITY_IMPORTANCE = 0.;
+    static const auto ENEMY_REACTION_TIME = Duration::fromSeconds(0);
```

Comment:
it seems like they could be tuned, so should these be dynamic parameters?

Reply:
It would be a pretty big change to dependency inject `PassingConfig` down to here, and it's very easy to tune these parameters quickly with simulated tests, the notebook, and Python bindings. I discussed this with @akhilveeraghanta at some point, he can also chime in on this